app/resource.py

```diff
@@ -154,6 +154,11 @@ def build_generic_model(tags: dict, resource_dict: dict, region_name: str, resou
                         iops=iops,
                         throughput=throughput)
 
+    # Instance should be stopped without warning
+    @staticmethod
+    def generic_is_stoppable_without_warning(resource):
+        return resource.ec2_type == 'instance' and resource.stop_after == '' and resource.state == 'stopped'
```

Comment:
I believe this should be `running`?
```suggestion
        return resource.ec2_type == 'instance' and resource.stop_after == '' and resource.state == 'running'
```

The notify phase will encode a warning date in the tag.

![image](https://user-images.githubusercontent.com/5783817/195934970-4a7fa390-2452-45a2-89e1-9642147b85c2.png)

One thing that can happen is if the tag isn't set NagBot will generate a tag like ` (Nagbot: Warned on 2022-10-14)` that won't match `== ''`.  It would be more robust to use `parse_date_tag` and then check `expiry_date` and `on_weekends` similar to `is_stoppable`

Reply:
Nice catch Cody Ray!!